src/js/services/user-service.js

```diff
@@ -65,3 +67,30 @@ export function trackEvent(eventName, userProps) {
         return response;
     });
 }
+
+export function updateNowViewing(deviceId) {
+    if (!deviceUpdateThrottle) {
```

Comment:
It doesn't call immediateUpdate at the tail end of the throttle if there is an update queued.